stdpopsim/ext/selection.py

```diff
@@ -151,3 +190,195 @@ def __attrs_post_init__(self):
     @classmethod
     def op_id(cls, op):
         return cls.op_types.index(op)
+
+
+def selective_sweep(
+    single_site_id,
+    population_id,
+    mutation_time,
+    selection_coeff,
+    dominance_coeff=0.5,
+    start_time=None,
+    end_time=None,
+    min_freq_at_start=None,
+    min_freq_at_end=None,
+):
+    """
+    Creates a list of SLiM events corresponding to a selective sweep at a
+    single site (see :meth:`Contig.add_single_site`) that may subsequently be
+    passed to the SLiM simulation engine. The sequence of events is:
+
+        (1) At time `mutation_time`, a **neutral** mutation is
+            introduced into population `population_id`.
+
+        (2) At time `start_time`, the mutation becomes beneficial in population
+            `population_id`. If the frequency of the mutation in population
+            `population_id` is below `min_freq_at_start` in this generation, the
+            simulation will restart from `mutation_time`.
+
+        (3) At time `end_time`, the mutation reverts to neutrality in
+            population `population_id`. If the frequency of the mutation in
+            population `population_id` is below `min_freq_at_end` in this
+            generation, the simulation will restart from `mutation_time`.
+
+    If the mutation is lost at any point between `mutation_time` and
+    `end_time`, the simulation will restart from `mutation_time`.
+
+    During the sweep, the fitnesses of individuals in population
+    `population_id` that are homozygous or heterozygous for the beneficial
+    mutation are `1 + selection_coeff` and  `1 + selection_coeff *
+    dominance_coeff`, respectively. Fitnesses of individuals in other
```

Comment:
Are you sure? The way it works in SLiM is that genotypes `aa`, `Aa`, and `AA` have fitnesses `1`, `1+hs` and `1+s`, respectively.

Reply:
oh wait that's what this said, just not in the order I expected, lol =)